scripts/change_alias.py

```diff
@@ -10,57 +41,117 @@ class ChangeAliases:
+                            logger.info(f"Alias {alias} has already been set to {current_alias}, nothing changed")
+                            index_to_match.pop(index)
+                            continue
+                        # not then remove the old alias
+                        else:
+                            remove_action = {"index": current_alias, "alias": alias}
+                            actions.append({"remove": remove_action})
+                    add_action = {"index": index, "alias": alias}
+                    actions.append({"add": add_action})
+                    index_to_match.pop(index)
+            # no actions to be taken, quit
+            if not actions:
+                # no actions could be two scenarios 1) already pointed to the wanted indices
+                # 2) wanted indices not existing
+                if index_to_match:
+                    print(f"No matching existing indices have been found, please use command "
+                        f"'curl {self.hosts[0]}/_cat/indices?v' to check all existing indices")
+                exit()
+            # use API to update aliases
+            payload = {"actions": actions}
+            self.es.indices.update_aliases(body=payload)
+
+            if index_to_match:
+                logger.warning("The following alias(es) has/ve not changed")
+                for curr in index_to_match.keys():
+                    logger.warning("alias " + index_to_match[curr] + " could not find the expected index " + curr)
+            else:
+                logger.info(f"Aliases have been updated using prefix {self.es_index_prefix}")
+                logger.info("New aliases are as following")
+                self.get_current_aliases()
+                for alias, index in self.current_aliases.items():
+                    logger.info("{} -> {}".format(alias, index))
 
-    def change_aliases(self, new_index, old_index, alias_name):
-        actions = {"actions": [
-            {"remove": {"index": old_index, "alias": alias_name}},
-            {"add": {"index": new_index, "alias": alias_name}},
-        ]}
-        self.es_staging.indices.update_aliases(body=actions)
+    def get_all_indices(self):
+        """
+        retrieve all indices from given Elastic Search
+        :return: list of names of all indices
+        """
+        logger.info("Get all indices")
+        result = self.es.indices.get_alias("*")
+        return list(result.keys())
 
+    def get_current_aliases(self):
+        """
+        This function will pring current aliases in format 'index_name' -> 'alias_name'
+        :param es_staging: staging elasticsearch object
+        :return: name of the current prefix or suffix in use
+        """
+        # could not use the method below which raises error when one specified alias does not exist,
+        # e.g. the pointed index gets deleted
+        # result = self.es.indices.get_alias(name=','.join(INDICES))
```

Comment:
Comments should be changed on "what are you doing here" and not "why I am not doing it this way"

Reply:
without such reasoning comment, people may switch back to the wrong way